### PR TITLE
MULE-18172: ObjectStoreToMapAdapter throws exceptions when used concurrently

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/test/java/org/mule/runtime/api/store/ObjectStoreToMapAdapterTestCase.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+    // Signal the threads to start processing.
+    startProcessingSemaphore.release(threadsSum);
+
+    // Wait for threads completion.
+    for (Thread thread : runningThreads) {
+      thread.join();
+    }
+
+    Assert.assertThat(numberOfExceptions.get(), is(0));
```
Comment (elrodro83) [inline src/test/java/org/mule/runtime/api/store/ObjectStoreToMapAdapterTestCase.java]: static import

File: src/test/java/org/mule/runtime/api/store/ObjectStoreToMapAdapterTestCase.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  }
+
+  @Test
+  public void containsKeyForKeyWithEquivalentHash() {
+    map.put("Ea", 5);
+    assertThat(map.containsKey("FB"), is(false));
+  }
+
+  @Test
+  public void getDoesntThrowExceptionsWhenKeyIsUpdatedConcurrently() throws InterruptedException {
```
Comment (elrodro83) [inline src/test/java/org/mule/runtime/api/store/ObjectStoreToMapAdapterTestCase.java]: `@Issue`

File: src/main/java/org/mule/runtime/api/store/ObjectStoreToMapAdapter.java [hunk 3]
```diff
@@ -91,6 +94,12 @@ public T put(String key, T value) {
         getObjectStore().store(key, value);
       }
       return previousValue;
+    } catch (ObjectDoesNotExistException | ObjectAlreadyExistsException e) {
+      // We should ignore these exceptions here, because an object could be:
+      // * Removed concurrently between contains and retrieve.
+      // * Removed between contains and remove.
+      // * Added between contains and store.
+      return null;
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/store/ObjectStoreToMapAdapter.java]: describe this behavior in the class level javadoc

File: src/main/java/org/mule/runtime/api/store/ObjectStoreToMapAdapter.java [hunk 4]
```diff
@@ -102,6 +111,9 @@ public T remove(Object key) {
       if (getObjectStore().contains((String) key)) {
         return getObjectStore().remove((String) key);
       }
+    } catch (ObjectDoesNotExistException e) {
+      // Object could be removed concurrently.
+      return null;
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/store/ObjectStoreToMapAdapter.java]: describe this behavior in the class level javadoc

**Comment by eze210** [comment]
--validate

**Pull request merged**